internal/resources/azure/signalr_service.go

```diff
@@ -0,0 +1,116 @@
+package azure
+
+import (
+	"fmt"
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"strings"
+)
+
+// SignalRService struct represents an Azure SignalR Service.
+//
+// Resource information: https://azure.microsoft.com/en-us/products/signalr-service
+// Pricing information: https://azure.microsoft.com/en-us/pricing/details/signalr-service/
+type SignalRService struct {
+	Address     string
+	Region      string
+	SkuName     string
+	SkuCapacity int64
+
+	MonthlyAdditionalMessages *int64 `infracost_usage:"monthly_additional_messages"`
+}
+
+// CoreType returns the name of this resource type
+func (r *SignalRService) CoreType() string {
+	return "SignalRService"
+}
+
+// UsageSchema defines a list which represents the usage schema of SignalRService.
+func (r *SignalRService) UsageSchema() []*schema.UsageItem {
+	return []*schema.UsageItem{
+		{Key: "monthly_additional_messages", DefaultValue: 0, ValueType: schema.Int64},
+	}
+}
+
+// PopulateUsage parses the u schema.UsageData into the SignalRService.
+// It uses the `infracost_usage` struct tags to populate data into the SignalRService.
+func (r *SignalRService) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+// BuildResource builds a schema.Resource from a valid SignalRService struct.
+// This method is called after the resource is initialised by an IaC provider.
+// See providers folder for more information.
+func (r *SignalRService) BuildResource() *schema.Resource {
+	// normalize sku to first letter capitalized
+	sku := cases.Title(language.English).String(strings.ToLower(r.SkuName))
```

Comment:
It might be a great candidate to become a new helper function. I see it's used in several places already. :)